packages/mongodb-access-adapter/datagrip-access-adapter/src/main/kotlin/com/mongodb/jbplugin/accessadapter/datagrip/adapter/DataGripMongoDbDriver.kt

```diff
@@ -33,60 +37,90 @@ import kotlinx.coroutines.withTimeout
+    private val dataSource: LocalDataSource,
 ) : MongoDbDriver {
     private val gson = Gson()
+    private val jsonWriterSettings = JsonWriterSettings.builder()
+.outputMode(JsonMode.EXTENDED)
+.indent(false)
+.build()
+
+    private fun Bson.toJson(): String = this.toBsonDocument().toJson(
+            jsonWriterSettings,
+        )
+
+    override suspend fun serverUri(): URI = URI.create(dataSource.url!!)
 
     override suspend fun <T : Any> runCommand(
-        command: Document,
+        command: Bson,
         result: KClass<T>,
-        timeout: Duration
-    ): T = withContext(
-        Dispatchers.IO
-    ) {
-        runQuery(
-            """db.runCommand(${command.toJson()})""",
-            result,
-            timeout
-        )[0]
-    }
+        timeout: Duration,
+    ): T =
+        withContext(
+            Dispatchers.IO,
+        ) {
+            runQuery(
+                """db.runCommand(${command.toJson()})""",
+                result,
+                timeout,
+            )[0]
+        }
 
     override suspend fun <T : Any> findOne(
         namespace: Namespace,
-        query: Document,
-        options: Document,
+        query: Bson,
+        options: Bson,
         result: KClass<T>,
-        timeout: Duration
-    ): T? = withContext(Dispatchers.IO) {
-        runQuery(
-            """db.getSiblingDB("${namespace.database}")
+        timeout: Duration,
+    ): T? =
+        withContext(Dispatchers.IO) {
+            runQuery(
+                """db.getSiblingDB("${namespace.database}")
                  .getCollection("${namespace.collection}")
-                 .findOne(${query.toJson()}, ${options.toJson()}) """.trimMargin(),
-            result,
-            timeout
-        ).getOrNull(0)
-    }
+                 .findOne(${query.toJson()}, ${options.toJson()}) 
+                """.trimMargin(),
+                result,
+                timeout,
+            ).getOrNull(0)
+        }
 
     override suspend fun <T : Any> findAll(
         namespace: Namespace,
-        query: Document,
+        query: Bson,
         result: KClass<T>,
         limit: Int,
-        timeout: Duration
+        timeout: Duration,
     ) = withContext(Dispatchers.IO) {
         runQuery(
             """db.getSiblingDB("${namespace.database}")
                  .getCollection("${namespace.collection}")
-                 .find(${query.toJson()}).limit($limit) """.trimMargin(),
+                 .find(${query.toJson()}).limit($limit) 
+            """.trimMargin(),
             result,
-            timeout
+            timeout,
         )
     }
 
+    override suspend fun countAll(
+        namespace: Namespace,
+        query: Bson,
+        timeout: Duration,
+    ) = withContext(Dispatchers.IO) {
+        runQuery(
+            """
+            db.getSiblingDB("${namespace.database}")
+                 .getCollection("${namespace.collection}")
+                 .countDocuments(${query.toJson()})
```

Comment:
Do we want a TODO ticket for adding proper escaping everywhere?

Reply:
Oh no, I just forgot about it (shame on me). Fixing it.